includes/class-wc-session-handler.php

```diff
@@ -63,6 +66,7 @@ public function __construct() {
 		}
 
 		$this->_data = $this->get_session_data();
+		$this->_data_org = serialize($this->_data);
```

Comment:
This is because I'm not sure what is in the $_data property.

It should be an array, so I could use ==. But maybe there are objects deeper in the array, these should also be deep compared. I'm not sure if == does this correctly.

If it does, then there is the problem of references. If I copy the $_data array in $_data_org and it's a shallow copy, the objects in the two arrays could be the same instance, then it will be impossible to determine if the value to save changed from the original value.

I decided to bypass all of this and try to stick as near to the database representation as I could (comfortably).
So I reduced the above complexity to just comparing two strings, the old one and the one to be saved.
I hope this is easier for programmers to understand, then to be able to reason that the deep compare of an array with values or objects of arbitrary depth will return the correct value in all possible permutations.
At least for me it's easier to comprehend.